server/plugin.go

```diff
@@ -28,102 +29,88 @@ func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Req
 	p.API.LogDebug("HTTP request", "Host", r.Host, "RequestURI", r.RequestURI, "Method", r.Method, "query", r.URL.Query().Encode())
 
 	config := p.getConfiguration()
-	if !config.Enabled || config.Secret == "" || config.UserName == "" {
-		errorMessage := "This plugin is not configured"
-		p.postHTTPDebugMessage(errorMessage)
-		http.Error(w, errorMessage, http.StatusForbidden)
+	if config.Secret == "" || config.UserName == "" {
+		http.Error(w, p.debug("JIRA plugin not configured correctly; must provide Secret and UserName"), http.StatusForbidden)
```

Comment:
Maybe we can just warp this whole function into a new one `handleRequest(r *http.Request) (int, error)` and get rid of this duplicate code pieces?

0/5 on the function name
2/5 that the error handling is not optimal

Reply:
-negative -2/5 on breaking up the function, it's a good single-purpose webhook handler now. When we have more kinds of http messages, yes.
- strongly agreed that the error handling is not optimal, out of scope for now

Reply:
Apologies, completely misunderstood your comment about wrapping into a function; did that now, muchas gracias for the suggestion, much cleaner.